stat_var_renaming/stat_var_renaming.py

```diff
@@ -0,0 +1,631 @@
+    ['populationType','measuredProp', 'measurementQualifier',
+    'measurementDenominator', 'statType', 'scalingFactor']
+  for c in range(1, MAX_CONSTRAINTS + 1):
+    properties_of_statistical_variables.append(f"orig_p{c}")
+    properties_of_statistical_variables.append(f"orig_v{c}")
+
+  # See if there are any duplicates outside of this set.
+  bad_overlaps = overlapping_stat_var_names[
+      overlapping_stat_var_names.duplicated(
+        properties_of_statistical_variables, keep=False) == False]
+
+  # Print out error if there are real collisions.
+  bad_overlaps = bad_overlaps.sort_values(['v1', 'v2', 'v3'])
+  bad_overlaps_str = ""
+  for index, row in bad_overlaps.iterrows():
+    bad_overlaps_str += row_to_stat_var_mcf(row) + ","
+  assert bad_overlaps.shape[0] == 0, "Duplicate StatVars!: " + bad_overlaps
+
+  # No issues so remove these duplicate names.
+  return stat_vars.drop_duplicates("HumanReadableName")
+
+def build_original_constraints(row):
+  """ Builds list of original properties and constraints in MCF Format.
+  Helper method for output MCF generation from a statistical variable row. 
+
+  Args:
+    row -> Row of pandas dataframe with renamed constraint and value fields.
+
+  Returns:
+    Multiline string of constraints and values in the format of 
+      p1: dcs:v1 
+      p2: dcs:v2... 
+  """
+  constraints_text = ""
+  current_constraint = 1
+  while (next_constraint <= MAX_CONSTRAINTS
+        and not pd.isna(row[f"orig_p{next_constraint}"])):
+    constraints_text += f"{row[f'orig_p{next_constraint}']}:"
+    constraints_text += f"dcs:{row[f'orig_v{next_constraint}']}\n"}:"
+    current_constraint += 1
+  return constraints_text
+
+def row_to_stat_var_mcf(row):
+  """ Creates MCF Statistical Variable node from a dataframe row.
+    
+  To integrate these new statistical variables into the datacommons graph,
+  they need to uploaded as MCF nodes with a new dcid, but the same properties.
+  This function creates that MCF as a text file from a row of the dataframe.
+
+  Args:
+    row -> Statistical variable dataframe row with HumanReadableName.
+  Returns:
+    Multiline string of the new MCF node for that statistical variable.
+  """
+  new_stat_var = (TEMPLATE_STAT_VAR
+    .replace("{human_readable_dcid}", row['HumanReadableName'])\
+    .replace("{populationType}", row['orig_populationType'])\
+    .replace("{statType}", row['statType'])\
+    .replace("{measuredProperty}", row['measuredProp'])\
+    .replace("{CONSTRAINTS}", build_original_constraints(row)))
+
+  # Add optional fields.
+  if not pd.isna(row['measurementQualifier']):
+    new_stat_var += f"measurementQualifier: dcs:{row['measurementQualifier']}\n"  
+  if not pd.isna(row['measurementDenominator']):
+    new_stat_var += (
+      f"measurementDenominator: dcs:{row['measurementDenominator']}\n")
+
+  return new_stat_var
+
+def remove_new_stat_vars(stat_vars, client):
+  """ Removes Statistical Variables that do not already exist. 
```

Comment:
This func might not stay--we do garbage collection on old BQ tables.

Reply:
I added it as optional. I have used it a few times in the past even if should not be run every time.